server/src/test/java/org/apache/druid/segment/metadata/CoordinatorSegmentMetadataCacheTest.java

```diff
@@ -2214,6 +2215,77 @@ protected void coldDatasourceSchemaExec()
     Assert.assertEquals(0, latch.getCount());
   }
 
+  @Test
+  public void testTombstoneSegmentIsNotAdded() throws InterruptedException
```

Comment:
The test is trying ensure that tombstone segments are not added to the Broker and Coordinator cache instances. 
Refactoring the test wouldn't be very beneficial since,
- Timeline used in the two caches is different. 
- `org.apache.druid.segment.metadata.AbstractSegmentMetadataCache#getSegmentMetadataSnapshot` implementation differs between the two instances.